generators/aws-containers/aws-client.js

```diff
@@ -0,0 +1,299 @@
+}
+
+/**
+ * Get the credentials from the ~/.aws/credentials file using the AWS_PROFILE env var to get the profile.
+ *
+ * @param profile The AWS profile to get the credentials from. Default to 'default'
+ * @returns {Promise} Will resolve with no parameters if it succeeds, rejects with the error if it fails (no credentials found for given profile.
+ */
+function saveCredentialsInAWS(profile = 'default') {
+    credentials = new AWS.SharedIniFileCredentials({ profile });
+    return new Promise((resolve, reject) => credentials.refresh((err) => {
+        if (err) {
+            reject(err);
+        }
+        AWS.config.credentials = credentials;
+        resolve();
+    }));
+}
+
+/**
+ * Retrieve decoded information to authenticate to Docker with AWS credentials.
+ * @returns {Promise} Returns a promise that resolves when the informations are retrieved.
+ */
+function getDockerLogin() {
+    return spinner(new Promise((resolve, reject) => _getAuthorizationToken()
+        .then(authToken => sts.getCallerIdentity({}).promise()
+            .then((data) => {
+                const decoded = utils.decodeBase64(authToken.authorizationToken);
+                const splitResult = _.split(decoded, ':');
+                resolve({
+                    username: splitResult[0],
+                    password: splitResult[1],
+                    accountId: data.Account
+                });
+            })
+            .catch(() => reject(new Error('Couldn\'t retrieve the user informations'))))));
+}
+
+/**
+ * Fetch Authentication token from AWS to authenticate with Docker
+ * @returns {Promise} Returns a promise that resolves when the informations are retrieved.
+ * @private
+ */
+function _getAuthorizationToken() {
+    return spinner(new Promise((resolve, reject) =>
+        ECR.sdk.getAuthorizationToken({}).promise()
+            .then((data) => {
+                if (!_.has(data, 'authorizationData.0')) {
+                    reject(new Error('No authorization data found.'));
+                    return;
+                }
+                resolve(data.authorizationData[0]);
+            })));
+}
+
+/**
+ * Create a S3 Bucket in said region with said bucketBaseName.
+ * the bucketBaseName will be used to create a
+ * @param bucketName the name of the bucket to create.
+ * @param region the region to create the bucket in.
+ * @returns {Promise}
+ */
+function createS3Bucket(bucketName, region = DEFAULT_REGION) {
+    const createBuckerParams = {
+        Bucket: bucketName
+    };
+    return spinner(new Promise((resolve, reject) => s3.headBucket({
+        Bucket: bucketName
+    }).promise()
+        .catch((error) => {
+            if (error.code !== 'NotFound') {
+                reject(new Error(`The S3 Bucket ${chalk.bold(bucketName)} in region ${chalk.bold(region)} already exists and you don't have access to it. Error code: ${chalk.bold(error.code)}`));
+            }
+        })
+        .then(() =>
```

Comment:
Can you call `then` before `catch` please?

Reply:
It was originally like this but because of the way Promises behave we've had to put it first. If we put the `catch` `after` the `then` and the `s3.createBucket()` promise fails, it falls back to the `spinner()` which will change the expected behavior

Reply:
That's weird.